lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2411,7 +2410,7 @@ instance Buildable e => LiftHandler (ErrSelectForPayment e) where
                 , "transaction, then cancel the previous one first."
                 ]
         ErrSelectForPaymentTxTooLarge maxSize maxN  ->
-            apiError err400 TransactionIsTooBig $ mconcat
+            apiError err403 TransactionIsTooBig $ mconcat
```

Comment:
Not sure, I think this might actually be https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413

Reply:
Sounds relevant for this use case indeed.

Reply:
Judging by the error message I'd say yes and no.

Yes, in case where the metadata is too big. As error message suggests:
```
Alternatively, try sending to less recipients or with smaller metadata.
```

No, if the transaction turns out to be too big:
```
Note that I am selecting inputs randomly, so retrying *may work* 
```

For the sake of _keeping it simple_ I'd stay with `403`, which is more general.